examples/graphs/node2vec_movielens.py

```diff
@@ -0,0 +1,563 @@
+
+    return np.array(targets), np.array(contexts), np.array(labels), np.array(weights)
+
+
+num_negative_smaples = 4
+targets, contexts, labels, weights = generate_examples(
+    sequences=walks,
+    window_size=num_steps,
+    num_negative_smaples=num_negative_smaples,
+    vocabulary_size=len(vocabulary),
+)
+
+"""
+Let's display the shapes of the outputs
+"""
+
+print(f"Targets shape: {targets.shape}")
+print(f"Contexts shape: {contexts.shape}")
+print(f"Labels shape: {labels.shape}")
+print(f"Weights shape: {weights.shape}")
+
+"""
+### Create tf.data.Dataset
+"""
+
+batch_size = 1024
+
+
+def create_dataset(targets, contexts, labels, weights, batch_size):
+    inputs = {
+        "target": targets,
+        "context": contexts,
+    }
+    dataset = tf.data.Dataset.from_tensor_slices((inputs, labels, weights))
+    dataset = dataset.shuffle(buffer_size=batch_size * 2)
+    dataset = dataset.batch(batch_size, drop_remainder=True)
+    return dataset
+
+
+dataset = create_dataset(
+    targets=targets,
+    contexts=contexts,
+    labels=labels,
+    weights=weights,
+    batch_size=batch_size,
+)
+
+"""
+## Train a Skip-gram Model
+
+Our skip-gram is a simple binary classification model that works as follows:
+1. An embedding is looked up for the `target` movie.
+2. An embedding is looked up for the `context` movie.
+3. The dot product is computed between these two embeddings.
+4. The result (after a sigmoid activation) is compared to the label.
+5. A binary crossentropy loss is used.
+"""
+
+learning_rate = 0.001
+embedding_dim = 50
+num_epochs = 10
+
+"""
+### Implement the model
+"""
+
+
+def create_model(vocabulary_size, embedding_dim):
+
+    inputs = {
+        "target": layers.Input(name="target", shape=(), dtype=tf.int32),
+        "context": layers.Input(name="context", shape=(), dtype=tf.int32),
+    }
+    # Initialize item embeddings.
+    embed_item = layers.Embedding(
+        input_dim=vocabulary_size,
+        output_dim=embedding_dim,
+        embeddings_initializer="he_normal",
+        embeddings_regularizer=keras.regularizers.l2(1e-6),
+        name="item_embeddings",
+    )
+    # Lookup embeddings for target.
+    target_embeddings = embed_item(inputs["target"])
+    # Lookup embeddings for context.
+    context_embeddings = embed_item(inputs["context"])
+    # Compute dot similarity between target and context embeddings.
+    logits = layers.Dot(axes=1, name="dot_similarity")(
+        [target_embeddings, context_embeddings]
+    )
+    # Create the model.
+    model = keras.Model(inputs=inputs, outputs=logits)
+    return model
+
+
+"""
+### Train the model
+"""
+
+model = create_model(len(vocabulary), embedding_dim)
+model.compile(
+    optimizer=keras.optimizers.Adam(learning_rate),
+    loss=keras.losses.BinaryCrossentropy(from_logits=True),
+)
+
+tf.keras.utils.plot_model(
+    model, show_shapes=True, show_dtype=True, show_layer_names=True,
+)
+
+history = model.fit(dataset, epochs=num_epochs)
```

Comment:
We should monitor generalization via a validation set during training.

Reply:
I might be wrong, but it seems think in self-supervised learning tasks/phases, having a validation set is not common. see for example [word2vec](https://www.tensorflow.org/tutorials/text/word2vec). 

WDYT?